undertow/WFLY-17273_Global_tracking_mode_setting_in_Undertow.adoc

```diff
@@ -0,0 +1,100 @@
+= WFLY-17273 Global tracking-mode setting in Undertow
+:author:            Lin Gao
+:email:             lgao@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+There is an element: `tracking-mode` in `session-config` section of web descriptor, which is used to specify the session tracking mode, the available value can be: `COOKIE`, `URL`, `SSL`. Example in the web.xml is like:
+
+[source,xml]
+----
+<session-config>
+    <tracking-mode>COOKIE</tracking-mode>
+    <tracking-mode>URL</tracking-mode>
+</session-config>
+----
+
+When it is not defined, undertow uses `COOKIE` and falls over to `URL` if the browser does not allow cookie for the deployed web applications.
+
+When users want to have different set up for all web applications, they need to update the web.xml in each application. It would be great if undertow subsystem supports specifying the session tracking modes globally at the servlet container level.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17273[WFLY-17273]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1980[EAP7-1980]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+WildFly Undertow subsystem
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+Provide a way to specify global session tracking modes at servlet container level. The configuration takes effect only if there are no session tracking modes specified in the deployment. So the configuration either in `web.xml` or updated in a `ServletContextListener` implementation takes precedence, which keeps consistent behavior as what it is now.
+
+=== Hard Requirements
+
+There will be 2 ways to set up global session tracking modes in the server:
+
+- Use system property with prefix `undertow.session.tracking-modes`. It reads from system property of `undertow.session.tracking-modes.CONTAINER-NAME`, and falls over to `undertow.session.tracking-modes` if it is not defined.
+* The value gets validated on deployment time, the benefit is that users can use `/system-property=*` to update the global session tracking modes after server starts up, the disadvantage is that the wrong configuration can only be noticed on the deployment time.
+* Use system property approach makes it possible to back port to downstream which does not allow model changes.
+
+- Use an attribute in servlet container: `/subsystem=undertow/servlet-container=default:write-attribute(name="session-tracking-modes", value=[COOKIE, URL])`
```

Comment:
I think that calling it defaullt-tracking-mode makes it clearer the purpose of the new attribute

Reply:
yes, `defaullt-tracking-mode` makes more sense